### ISSUE TITLE
[pytorch] fully_shard(reshard_after_forward=) changes in next release

### ISSUE THREAD

**Issue by weifengpy** [volcengine/verl#1894]
Dear Verl engs, 

Greetings from pytorch fsdp2. I am landing a "bc-breaking" change into pytorch nightly for `fully_shard(reshard_after_forward=)`: https://github.com/pytorch/pytorch/pull/155319

without my change
* root model will override reshard_after_forward to False, no matter user config it as reshard_after_forward=True/False, or did not specify reshard_after_forward

after my change
* if user specify reshard_after_forward, we will respect it
* if user didn't specify reshard_after_forward, we use reshard_after_forward=False for root, and reshard_after_forward=True

what it means for verl
https://github.com/volcengine/verl/blob/c0f5ccbe5da32ee0acfc60dfe44658b3da828792/verl/utils/fsdp_utils.py#L443-L445

* option 1: explicitly set ``fsdp_kwargs["reshard_after_forward"] = False`` for root model (because root will be used immeidately in backward
* option 2: expose a config for user to decide. without full AC, the peak memory might happen during cross entropy loss. `reshard_after_forward=True/False` becomes a memory/perf trade off

**Comment by weifengpy** [comment]
cc @lxg2015 @eric-haibin-lin for upcoming fsdp2 "bc-breaking" changes

**Comment by Kirrito-k423** [comment]
定时清理检查结论：根据该 issue 的 comments 与 timeline，最近一次非当前运行账号的公开回复或公开引用是 `2025-06-06T20:59:37Z`，来源为 `mentioned`（作者 `eric-haibin-lin`，链接：https://github.com/verl-project/verl/issues/1894）。这个时间距离当前定时任务运行日期 2026-07-07 已超过 6 个月。

因此按照社区 issue 定时清理标准关闭此 issue。检查来源包括该 issue 的公开页面 https://github.com/verl-project/verl/issues/1894、comments 列表和 timeline 事件。如果作者或社区成员仍能提供新的复现、证据或需求背景，欢迎重新评论请求重开，或创建新的 issue 并引用本 issue。

---

Scheduled cleanup check: based on this issue's comments and timeline, the latest public reply or public reference from an account other than the current automation account was `2025-06-06T20:59:37Z`. The source was `mentioned` by `eric-haibin-lin`: https://github.com/verl-project/verl/issues/1894. This is more than 6 months before the scheduled run date, 2026-07-07.

Therefore, this issue is being closed under the scheduled issue cleanup policy. The checked sources include the public issue page https://github.com/verl-project/verl/issues/1894, the comments list, and timeline events. If the author or community can still provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or file a new issue referencing this one.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
